events/reporter.go

```diff
@@ -102,18 +127,16 @@ func ReportRewardReceived(r Reward) {
 		SmesherID: r.Smesher.ToBytes(),
 	})
 
-	if reporter != nil {
-		if reporter.blocking {
-			reporter.channelReward <- r
-			log.Debug("reported reward: %v", r)
-		} else {
+	if reporter != nil && len(reporter.rewardsSubs) > 0 {
```

Comment:
You can remove the second part of the check here, `len(reporter.rewardsSubs) > 0` - this is implicit in the loop